@xen-orchestra/lite/src/components/RemoteConsole.vue

```diff
@@ -3,42 +3,96 @@
+const FIBONACCI_MS_ARRAY: number[] = Array.from(
+  fibonacci().toMs().take(N_TOTAL_TRIES)
+);
+const INITIAL_VALUE_N_CONNECTION_ATTEMPTS = 0;
+
 const props = defineProps<{
   location: string;
+  isConsoleAvailable: boolean;
 }>();
 
 const vmConsoleContainer = ref<HTMLDivElement>();
 const xenApiStore = useXenApiStore();
-let vncClient: VncClient | undefined;
-
-watchEffect(() => {
-  if (!vmConsoleContainer.value || !xenApiStore.currentSessionId) {
+const url = computed(() => {
+  if (xenApiStore.currentSessionId === null) {
     return;
   }
+  const _url = new URL(props.location);
+  _url.protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
+  _url.searchParams.set("session_id", xenApiStore.currentSessionId);
+  return _url;
+});
 
-  if (vncClient !== undefined) {
-    vncClient.disconnect();
-    vncClient = undefined;
+let vncClient: VncClient | undefined;
+let nConnectionAttempts = INITIAL_VALUE_N_CONNECTION_ATTEMPTS;
+
+const handleDisconnectionEvent = () => {
+  clearVncClient();
+  if (props.isConsoleAvailable) {
+    console.error(
+      `Connection lost for the remote console: ${
+        props.location
+      }. New attempt in ${FIBONACCI_MS_ARRAY[nConnectionAttempts - 1]}ms`
+    );
+    createVncConnection();
   }
+};
+const handleConnectionEvent = () =>
+  (nConnectionAttempts = INITIAL_VALUE_N_CONNECTION_ATTEMPTS);
 
-  const url = new URL(props.location);
-  url.protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
-  url.searchParams.set("session_id", xenApiStore.currentSessionId);
+const clearVncClient = () => {
+  vncClient?.removeEventListener("disconnect", handleDisconnectionEvent);
+  vncClient?.removeEventListener("connect", handleConnectionEvent);
+  if (vncClient?._rfbConnectionState !== "disconnected") {
+    vncClient?.disconnect();
+  }
+  vncClient = undefined;
+};
+const createVncConnection = async () => {
+  if (nConnectionAttempts > N_TOTAL_TRIES) {
+    console.error(
+      "The number of reconnection attempts has been exceeded for:",
+      props.location
+    );
+    return;
+  }
+  await new Promise((resolve) =>
+    setTimeout(resolve, FIBONACCI_MS_ARRAY[nConnectionAttempts++ - 1])
+  );
 
-  vncClient = new VncClient(vmConsoleContainer.value, url.toString(), {
+  vncClient = new VncClient(vmConsoleContainer.value!, url.value!.toString(), {
     wsProtocols: ["binary"],
   });
-
   vncClient.scaleViewport = true;
+
+  vncClient.addEventListener("disconnect", handleDisconnectionEvent);
+  vncClient.addEventListener("connect", handleConnectionEvent);
+};
+
+watch(url, clearVncClient);
+watchEffect(() => {
+  if (
+    url.value === undefined ||
+    vmConsoleContainer.value === undefined ||
+    !props.isConsoleAvailable
+  ) {
+    return;
+  }
+
+  nConnectionAttempts = INITIAL_VALUE_N_CONNECTION_ATTEMPTS;
+  createVncConnection();
 });
 
 onBeforeUnmount(() => {
-  vncClient?.disconnect();
-  vncClient = undefined;
+  clearVncClient();
 });
 </script>
```

Comment:
I don't understand the whole change of this file.

I just tried a reboot on the previous version then on this version, and I don't see any difference 🤔 

### Here is the previous version:

https://user-images.githubusercontent.com/19408/222766800-dcc300e9-7f49-4f63-a0da-278e7a5c2970.mp4

### Here is the current version:

https://user-images.githubusercontent.com/19408/222766821-89c679a7-dc55-4716-8543-cf9b4b8ad942.mp4

Reply:
The changes inside `RemoteConsole` are mainly made to handle the case of an unintentional loss of connection.
The reboot feature is handled by the `isConsoleAvailable` prop